error.go

```diff
@@ -55,7 +58,7 @@ func NewError(apiName string, handle interface{}) error {
 			break
 		}
 		if IsError(ret) {
-			panic(fmt.Errorf("SQLGetDiagRec failed: ret=%d", ret))
+			return fmt.Errorf("SQLGetDiagRec failed: ret=%d", ret)
```

Comment:
NewError function is called when `handle` parameter has outstanding error condition on it. The purpose of NewError is to retrieve that error and return it to the caller. But you are replacing that error with completely new error returned from SQLGetDiagRec. Why it is OK to silently replace the error? What about original error? Wouldn't caller need to know what the original error was?